File: 2211-k-radius-subarray-averages/k-radius-subarray-averages.cpp
class Solution {
public:
    vector<int> getAverages(vector<int>& nums, int k) {
        vector<int> ans(nums.size(), -1);
        long long sum = 0;
        int i=0, j=0, idx = k;
        while (j<nums.size()){
            sum+=nums[j];
            if (j-i+1 < 2*k+1){
                j+=1;
            } else if (j-i+1 == 2*k+1){
                int avg = sum/(2*k+1);
                ans[idx] = avg;
                idx+=1;
                j+=1;
                sum-=nums[i];
                i+=1;
            }
        }
        return ans;
    }
};

File: 2211-k-radius-subarray-averages/k-radius-subarray-averages.go
func getAverages(nums []int, k int) []int {
    radius := make([]int, len(nums))
    for i := range radius {
        radius[i] = -1
    }

    for i := 1; i < len(nums); i++ {
        nums[i] += nums[i-1]
    }

    for i := k; i < len(nums) - k; i++ {
        n := 2 * k + 1
        sum := nums[i+k]

        if i > k {
            sum -= nums[i-k-1]
        }

        average := sum / n
        radius[i] = average
    }

    return radius
}